.github/workflows/build.yml

```diff
@@ -33,3 +33,18 @@ jobs:
 
     - name: Run tests
       run: nix flake -L check
+
+    - name: Check build without nix environment
+      run: |
+        nix develop -c "env" | grep CMAKE_PREFIX_PATH >cmake_prefix
+        echo "export `cat cmake_prefix`" > cmake_prefix
+        source cmake_prefix
+        echo $CMAKE_PREFIX_PATH
+        CMAKE_BINARY=$(nix develop -c bash -c 'which cmake' | tail -1)
+        echo $CMAKE_BINARY
+        NINJA_BINARY=$(nix develop -c bash -c 'which ninja' | tail -1)
+        echo $NINJA_BINARY
+        CXX=$(nix develop -c bash -c 'which g++' | tail -1)
+        echo $CXX
+        $CMAKE_BINARY -GNinja -DCMAKE_MAKE_PROGRAM=$NINJA_BINARY -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER=$CXX -DBUILD_ASSIGNER_TESTS=TRUE
+        $NINJA_BINARY -C build
```

Comment:
How about we move this into some shell script like `scripts/non-nix-build.sh` so one can run this locally?

Reply:
Move to script as is. I was thinking about a customizable script, but there are too many parameters, so it's hard to make the script useful.